cmd/gapid/main.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+const versionPrefix = `version "`
+
+func main() {
+	cwd, err := filepath.Abs(filepath.Dir(os.Args[0]))
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	cwd, err = filepath.EvalSymlinks(cwd)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	args := os.Args[1:]
+	gapicArgs := make([]string, 0, len(args))
+	vm := ""
+	vmargs := make([]string, 0)
+	help := false
+	console := false
+
+	for i := 0; i < len(args); i++ {
+		switch {
+		case args[i] == "--vm" && i < len(args)-1:
+			i = i + 1
+			vm = args[i]
+		case args[i] == "--vmarg" && i < len(args)-1:
+			i = i + 1
+			vmargs = append(vmargs, args[i])
+		case args[i] == "--console":
+			console = true
+			createConsole()
+		default:
+			help = help || args[i] == "--help"
+			gapicArgs = append(gapicArgs, args[i])
+		}
+	}
+
+	if help && !console {
```

Comment:
To be slightly safer in avoiding future changes that could cause `createConsole` to be called more than once, I'd recommend removing the call to `createConsole` above, setting `console = 
true` for `--help` and change this line to: 

```go
if console {
    createConsole()
}
```